MapleServer2/PacketHandlers/Game/DungeonHandler.cs

```diff
@@ -52,8 +59,60 @@ public override void Handle(GameSession session, PacketReader packet)
         public static void HandleEnterDungeonRoom(GameSession session, PacketReader packet)
         {
             int dungeonId = packet.ReadInt();
+            bool groupEnter = packet.ReadBool();
+            Player player = session.Player;
+
+            if (player.DungeonSessionId != -1)
+            {
+                session.SendNotice("Leave your current dungeon before opening another.");
+                return;
+            }
+
+            int dungeonLobbyId = DungeonStorage.GetDungeonByDungeonId(dungeonId).LobbyFieldId;
+            MapPlayerSpawn spawn = MapEntityStorage.GetRandomPlayerSpawn(dungeonLobbyId);
+
+            DungeonSession dungeonSession = GameServer.DungeonManager.CreateDungeonSession(dungeonId, groupEnter ? DungeonType.group : DungeonType.solo);
+            session.SendNotice($"dungeon session created sessionID:{dungeonSession.SessionId} instanceId: {dungeonSession.DungeonInstanceId}");
+
+            //Todo: Send packet that greys out enter alone / enter as party when already in a dungeon session.
+            if (groupEnter)
+            {
+
+                Party party = GameServer.PartyManager.GetPartyById(session.Player.PartyId);
+                foreach (Player member in party.Members)
+                {
+                    if (member.DungeonSessionId != -1)
+                    {
+                        session.SendNotice($"{member.Name} is still in a Dungeon Instance.");
+                        return;
+                    }
+                }
+                if (party.DungeonSessionId != -1)
+                {
+                    session.SendNotice("Need to reset dungeon before entering another instance");
+                    return;
+                }
+                //group takes dungeonsession of leader, because this packet is coming from leader.
+                party.DungeonSessionId = dungeonSession.SessionId;
+                //party.BroadcastParty(session => session.Send(PartyPacket.UpdatePlayer(session.Player)));
+                //party.BroadcastPacketParty(PartyPacket.UpdatePlayer(session.Player));
+                //party.BroadcastPacketParty(PartyPacket.PartyHelp(dungeonId, 1));
+                //party.BroadcastPacketParty(DungeonWaitPacket.Show(dungeonId, DungeonStorage.GetDungeonByDungeonId(dungeonId).MaxUserCount));
+            }
+            if (!groupEnter)
+            {
+                player.DungeonSessionId = dungeonSession.SessionId;
+            }
+            session.Player.Warp(spawn.Coord.ToFloat(), spawn.Rotation.ToFloat(), dungeonLobbyId, dungeonSession.DungeonInstanceId);
```

Comment:
This warp will only move the leader right? So will need something like 'partyWarp'

Reply:
Yea, only warps the leader, which is intended. 
In official maple story, players don't get automatically teleported in the new dungeon session.
They get the "enter dungeon" button under the party. I will handle this in a commit later.

Reply:
Ooh ok.